### PR TITLE
Use GitHub Actions and remove Travis

### PR DESCRIPTION
**Pull request by xalopp**
### Type of PR

* [ ] Bugfix
* [ ] New Feature
* [X] Other (explain):

Use GitHub Actions as CI 
### Breaking changes

* [ ] Yes, this is a breaking change

### Description
We want to use GitHub Actions in the future

### CODE REVIEW

**Comment by xalopp** [comment]
@mmoll  happy?

**Comment by xalopp** [comment]
@mmoll the slack integration isn't yet working because of [Settings - Actions secrets](https://github.com/mayflower/mo4-coding-standard/settings/secrets/actions):
```
Secrets are not passed to workflows that are triggered by a pull request from a fork. Learn more. 
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
+name: MO4 Coding Standard CI
+on:
+  pull_request:
+  push:
+  schedule:
+  - cron: '30 5 * * 1'
```
Comment (mmoll) [inline .github/workflows/ci.yml:6]: Feels a bit too often for me, how about this?
```suggestion
  - cron: '30 5 1 * *'
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      - run: xmllint --noout --schema vendor/squizlabs/php_codesniffer/phpcs.xsd phpcs.xml.dist
+      - run: diff -B MO4/ruleset.xml <(xmllint --format MO4/ruleset.xml)
+      - run: diff -B phpcs.mo4.xml <(xmllint --format phpcs.mo4.xml)
+      - run: diff -B phpcs.xml.dist <(xmllint --format phpcs.xml.dist)
+      - name: Stylecheck against MO4 itself
+        run: vendor/bin/phpcs
+  run-tests:
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
```
Comment (mmoll) [inline .github/workflows/ci.yml:41]: why not this?
```suggestion
        os: [ubuntu-latest]
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        run: vendor/bin/psalm --show-info=true
+      - name: Run phan
+        run: vendor/bin/phan
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          verbose: true
+
+  slack-notify:
```
Comment (mmoll) [inline .github/workflows/ci.yml:101]: Please remove this in this PR and add it in another PR from a branch in this very repo after this PR got merged, then the whole fork secret stuff shouldn't have any influence.

File: README.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
 # MO4 CodeSniffer ruleset
- 
+
 Provides a PHP CodeSniffer ruleset for the MO4 coding standard
 
-[![Build Status](https://travis-ci.org/mayflower/mo4-coding-standard.svg?branch=master)](https://travis-ci.org/mayflower/mo4-coding-standard)
+[![Build Status](https://github.com/mayflower/mo4-coding-standard/actions/workflows/ci.yml/badge.svg)](https://github.com/mayflower/mo4-coding-standard/actions)
```
Comment (mmoll) [inline README.md:5]: `.travis.yml` should also get deleted. 🗑️

File: .github/workflows/ci.yml (hunk 2 above)
Comment (xalopp) [reply .github/workflows/ci.yml:41]: Let's give it a try.. I picked the recommend version from the support matrix of `shivammathur/setup-php@v2` but the documented supported version of Windows also were not ... correct

File: .github/workflows/ci.yml (hunk 3 above)
Comment (xalopp) [reply .github/workflows/ci.yml]: Will do so

**Pull request merged**